src/aiida/engine/processes/calcjobs/manager.py

```diff
@@ -152,10 +152,14 @@ async def _update_job_info(self) -> None:
             raise
         else:
             for job_id in self._polling_jobs:
-                future = self._job_update_requests.pop(job_id)
-                if future.done():
-                    continue
-                future.set_result(self._jobs_cache.get(job_id, None))
+                future = self._job_update_requests.pop(job_id, None)
+                if future is None:
+                    self.logger.info(
```

Comment:
Great idea to add logging to this, thanks. I think this should be at least a `logger.warning` if not `logger.error` since our key assumption here has been violated.